### PR TITLE
Migrate acls to generics

### PR DESCRIPTION
**Pull request by schmichael**
See hashicorp/go-immutable-radix#43

Needs to wait until upstream is merged

### CODE REVIEW

**Comment by schmichael** [comment]
TODO: Seems to break [`stringer`](https://golang.org/x/tools/cmd/stringer) which we use to generate some code from `nomad/worker*.go`.

https://github.com/hashicorp/nomad/runs/7308742540?check_suite_focus=true#step:4:483

File: go.mod
Comment (shoenig) [inline go.mod:295]: whoops

**Review by shoenig — APPROVED** [review]

**Comment by shoenig** [comment]
@schmichael I rebased with main now that v2 is published; I leave it to you to give it a once-over && merge

**Pull request merged**
